app/components/QrCodeScanner/QrCodeScanner.jsx

```diff
@@ -29,24 +52,99 @@ export default class QrCodeScanner extends Component<Props> {
       callback(content, this.stopScanner.bind(this))
     })
 
-    Instascan.Camera.getCameras()
+    // since the browser halts while getting usermedia
+    // let the loading animation finish one round and only then continue
+    this.setState({ loading: true })
+    new Promise(resolve => setTimeout(resolve, ANIMATION_DURATION))
+      .then(() => Instascan.Camera.getCameras())
       .then((cameras: Array<Object>) => {
-        if (cameras.length > 0) {
-          this.scannerInstance.start(cameras[0])
-        } else {
-          console.error('No cameras found.')
+        if (cameras.length === 0) {
+          // shouldn't happen, case covered by withCameraAvailability
+          throw new Error('No cameras found.')
         }
+        return this.scannerInstance.start(cameras[0])
+      })
+      .catch(err => {
+        this.setState({ error: this.constructor.getScannerError(err) })
+      })
+      .finally(() => {
+        this.setState({ loading: false })
       })
-      .catch(e => console.error(e))
   }
 
   stopScanner() {
     if (this.scannerInstance) this.scannerInstance.stop()
+    this.setState({ error: null }) // clear error
+  }
+
+  static getScannerError(err: Error) {
+    const scanErr: ScannerError = {
+      message: 'Could not connect to camera'
+    }
+
+    if (err.name === 'TrackStartError') {
+      // get link info by user os, defaults to nothing
+      const docs = {
+        darwin: [
+          'https://support.apple.com/en-il/guide/mac-help/mh32356/10.14/mac',
```

Comment:
Way to put in the extra effort and track down these support articles... really nice touch 👍